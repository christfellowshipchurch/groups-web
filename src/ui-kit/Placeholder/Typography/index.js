import React, { useRef } from 'react';
import ReactPlaceholder from 'react-placeholder';
import PropTypes from 'prop-types';

import Line from '../Line';

// const boom = ({ theme, typographicStyles }) => {
//   const lineHeightIndex = typographicStyles.findIndex(
//     (part) => typeof part === 'string' && part.includes('line-height') // Ship it!
//   );
//
//   return typographicStyles[lineHeightIndex + 1]({ theme });
// };

// const TypographyLine = styled.default(Line).attrs({
//   className: 'h1 ui-kit Placeholder TypographyLine',
// })` height: ${boom}`;

const Typography = ({ children, isLoading, ...props }) => {
  const ref = useRef(children);

  return (
    <ReactPlaceholder
      customPlaceholder={
        <Line
          className={`${ref?.current?.type?.target} ui-kit-Placeholder-TypographyLine`}
        />
      }
      ready={!isLoading}
      {...props}
    >
      {children}
    </ReactPlaceholder>
  );
};

Typography.propTypes = {
  children: PropTypes.element.isRequired,
  isLoading: PropTypes.bool,
};

Typography.defaultProps = { isLoading: true };

Typography.displayName = 'ui-kit.Placeholder.Typography';

export default Typography;
