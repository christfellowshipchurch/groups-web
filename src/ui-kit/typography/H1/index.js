import React from 'react';
import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';

import { withOverrides } from '../../theme';
import { getIsLoading } from '../../isLoading';
import { Line } from '../../Placeholder';

const verticalRhythm = css`
  ${({ withMargins, theme }) =>
    withMargins ? `${theme.helpers.verticalRhythm(2.015625)} 0` : 'initial'};
`;

const StyledH1 = styled.h1.attrs(withOverrides('uiKit.typography.H1.StyledH1'))`
  font-size: 2.6875rem;
  font-weight: 900;
  line-height: ${({ theme }) => theme.helpers.verticalRhythm(2.6875, 1.15)};
  margin: ${verticalRhythm};

  /* Styles passed via the style prop and/or overrides */
  ${({ $style }) => $style}
`;

const Placeholder = styled(Line).attrs(
  withOverrides('uiKit.typography.H1.Placeholder')
)`
  height: 2.6875rem; /* Should match .h1 font-size */
  margin: ${({ withMargins, theme }) =>
    // if withMargins...
    withMargins
      ? // return rhythm so placeholder matches text position to prevent jitter/jank
        verticalRhythm
      : // else default to...
        `calc(${theme.helpers.verticalRhythm(
          2.6875,
          1.15
        )} - 2.6875rem) 0`}; /* line-height - font-size = space between lines */

  /* Styles passed via overrides */
  ${({ $style }) => $style}
`;

// We pass `style` as a prop to `StyledH2` so that custom styles are handled by styled components and not react (inline style)
const H1 = getIsLoading(
  ({ children, isLoading, withMargins, style, ...props }) => (
    <Placeholder
      isLoading={isLoading}
      withMargins={withMargins}
      showLoadingAnimation
    >
      <StyledH1 withMargins={withMargins} $style={style} {...props}>
        {children}
      </StyledH1>
    </Placeholder>
  )
);

H1.propTypes = {
  children: PropTypes.node.isRequired,
  isLoading: PropTypes.bool,
  withMargins: PropTypes.bool,
  style: PropTypes.any, // eslint-disable-line
};

H1.displayName = 'ui-kit.typography.H1';

export default styled(H1).attrs(withOverrides('uiKit.typography.H1'))``;
