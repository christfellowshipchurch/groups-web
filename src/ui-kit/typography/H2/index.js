import React from 'react';
import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';

import { Line } from '../../Placeholder';

const verticalRhythm = css`
  ${({ padded, theme }) =>
    padded
      ? `${theme.helpers.verticalRhythm(1)} 0 ${theme.helpers.verticalRhythm(
          0.834
        )}`
      : 'initial'};
`;

const StyledH2 = styled.h2.attrs({
  className: 'ui-kit-typography-H2-StyledH2',
})`
  font-size: 2.25rem;
  font-weight: 900;
  line-height: ${({ theme }) => theme.helpers.verticalRhythm(2.25, 1.15)};
  margin: ${verticalRhythm};
`;

const Placeholder = styled(Line).attrs({
  className: 'ui-kit-H1-Placeholder',
})`
  height: 2.25rem; /* Should match .h2font-size */
  margin: ${({ padded, theme }) =>
    // if padded...
    padded
      ? // return rhythm so placeholder matches text position to prevent jitter/jank
        verticalRhythm
      : // else default to...
        `calc(${theme.helpers.verticalRhythm(
          2.25,
          1.15
        )} - 2.25rem) 0`}; /* line-height - font-size = space between lines */
`;

const H2 = ({ children, isLoading, padded, ...props }) => (
  <Placeholder isLoading={isLoading} padded={padded} showLoadingAnimation>
    <StyledH2 padded={padded} {...props}>
      {children}
    </StyledH2>
  </Placeholder>
);

H2.defaultProps = {
  isLoading: false,
};

H2.propTypes = {
  children: PropTypes.node.isRequired,
  isLoading: PropTypes.bool,
  padded: PropTypes.bool,
};

H2.displayName = 'ui-kit.typography.H2';

export default H2;
