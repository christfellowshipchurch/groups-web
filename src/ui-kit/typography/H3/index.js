import React from 'react';
import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';

import Placeholder from '../../Placeholder';

const h3Style = css`
  h3,
  .H3 {
    color: ${({ theme }) => theme.colors.text.primary};
    font-family: ${({ theme }) => theme.typography.fontFamily};
    font-size: 1.5rem;
    font-weight: 900;
    line-height: ${({ theme }) => theme.helpers.verticalRhythm(1.5, 1.15)};
    &.custom-placeholder {
      height: 1.5rem; /* Should match .h3font-size */
    }
  }
`;

const StyledH3 = styled.h3.attrs({
  className: 'ui-kit-typography-H3-StyledH3',
})`
  margin: ${({ padded, theme }) =>
    padded
      ? `${theme.helpers.verticalRhythm(
          1.125
        )} 0 ${theme.helpers.verticalRhythm(0.75)} 0`
      : 'inital'};
`;

const H3 = ({ children, isLoading, padded, ...props }) => (
  <Placeholder.Typography
    isLoading={isLoading}
    showLoadingAnimation
    typographicClassName={'H3'} // needs to match child components className
  >
    <StyledH3 padded={padded} {...props}>
      {children}
    </StyledH3>
  </Placeholder.Typography>
);

H3.defaultProps = {
  isLoading: false,
};

H3.propTypes = {
  children: PropTypes.node.isRequired,
  isLoading: PropTypes.bool,
  padded: PropTypes.bool,
};

H3.displayName = 'ui-kit.typography.H3';

export { H3 as default, h3Style };
