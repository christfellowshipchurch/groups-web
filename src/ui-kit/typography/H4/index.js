import React from 'react';
import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';

import { Line } from '../../Placeholder';

const verticalRhythm = css`
  margin: ${({ padded, theme }) =>
    padded
      ? `${theme.helpers.verticalRhythm(0.66)} 0 ${theme.helpers.verticalRhythm(
          0.5
        )} 0`
      : 'initial'};
`;

const StyledH4 = styled.h4.attrs({
  className: 'ui-kit-typography-H4-StyledH4',
})`
  font-size: 1rem;
  font-weight: 700;
  line-height: ${({ theme }) => theme.helpers.verticalRhythm(1)};

  ${verticalRhythm}
`;

const Placeholder = styled(Line).attrs({
  className: 'ui-kit-H1-Placeholder',
})`
  height: 1rem; /* Should match H4 font-size */

  ${verticalRhythm}
`;

const H4 = ({ children, isLoading, padded, ...props }) => (
  <Placeholder isLoading={isLoading} padded={padded} showLoadingAnimation>
    <StyledH4 padded={padded} {...props}>
      {children}
    </StyledH4>
  </Placeholder>
);

H4.defaultProps = {
  isLoading: false,
};

H4.propTypes = {
  children: PropTypes.node.isRequired,
  isLoading: PropTypes.bool,
  padded: PropTypes.bool,
};

H4.displayName = 'ui-kit.typography.H4';

export default H4;
