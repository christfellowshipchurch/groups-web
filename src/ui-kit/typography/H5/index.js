import React from 'react';
import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';

import Placeholder from '../../Placeholder';

const h5Style = css`
  h5,
  .H5 {
    color: ${({ theme }) => theme.colors.text.primary};
    font-family: ${({ theme }) => theme.typography.fontFamily};
    font-size: 0.875rem;
    font-weight: 500;
    line-height: ${({ theme }) => theme.helpers.verticalRhythm(0.875)};
    &.custom-placeholder {
      height: 0.875rem; /* Should match .h5font-size */
    }
  }
`;

const StyledH5 = styled.h5.attrs({
  className: 'ui-kit-typography-H5-StyledH5',
})`
  margin: ${({ padded, theme }) =>
    padded
      ? `${theme.helpers.verticalRhythm(
          0.5775
        )} 0 ${theme.helpers.verticalRhythm(0.4375)} 0`
      : 'inital'};
`;

const H5 = ({ children, isLoading, padded, ...props }) => (
  <Placeholder.Typography
    isLoading={isLoading}
    showLoadingAnimation
    typographicClassName={'H5'} // needs to match child components className
  >
    <StyledH5 padded={padded} {...props}>
      {children}
    </StyledH5>
  </Placeholder.Typography>
);

H5.defaultProps = {
  isLoading: false,
};

H5.propTypes = {
  children: PropTypes.node.isRequired,
  isLoading: PropTypes.bool,
  padded: PropTypes.bool,
};

H5.displayName = 'ui-kit.typography.H5';

export { H5 as default, h5Style };
