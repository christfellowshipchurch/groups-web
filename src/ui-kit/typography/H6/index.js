import React from 'react';
import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';

import Placeholder from '../../Placeholder';

const h6Style = css`
  h6,
  .H6 {
    color: ${({ theme }) => theme.colors.text.secondary};
    font-family: ${({ theme }) => theme.typography.fontFamily};
    font-size: 0.75rem;
    font-weight: 500;
    line-height: ${({ theme }) => theme.helpers.verticalRhythm(0.75)};
    &.custom-placeholder {
      height: 0.75rem; /* Should match .h6font-size */
    }
  }
`;

const StyledH6 = styled.h6.attrs({
  className: 'ui-kit-typography-H6-StyledH6',
})`
  margin: ${({ padded, theme }) =>
    padded ? `0 0 ${theme.helpers.verticalRhythm(0.375)} 0` : 'initial'};
`;

const H6 = ({ children, isLoading, padded, ...props }) => (
  <Placeholder.Typography
    isLoading={isLoading}
    showLoadingAnimation
    typographicClassName={'H6'} // needs to match child components className
  >
    <StyledH6 padded={padded} {...props}>
      {children}
    </StyledH6>
  </Placeholder.Typography>
);

H6.defaultProps = {
  isLoading: false,
};

H6.propTypes = {
  children: PropTypes.node.isRequired,
  isLoading: PropTypes.bool,
  padded: PropTypes.bool,
};

H6.displayName = 'ui-kit.typography.H6';

export { H6 as default, h6Style };
