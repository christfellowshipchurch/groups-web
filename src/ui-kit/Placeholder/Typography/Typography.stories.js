import React from 'react';
import { storiesOf } from '@storybook/react';

import { H1, H2 } from '../../typography';

import Typography from '.';

storiesOf('ui-kit/Placeholder/Typography', module)
  .add('default', () => (
    <>
      <Typography typographicClassName={'h1'}>
        <H1>Boom</H1>
      </Typography>
      <Typography typographicClassName={'h1'}>
        <H2>Boom</H2>
      </Typography>
    </>
  ))
  .add('isLoading (false)', () => (
    <Typography isLoading={false}>
      <H1>Boom</H1>
    </Typography>
  ));
