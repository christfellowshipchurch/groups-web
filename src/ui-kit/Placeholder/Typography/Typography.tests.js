import React from 'react';
import renderer from 'react-test-renderer';

import { H1 } from '../../typography';
import Providers from '../../../Providers';

import Typography from '.';

describe('The Typography placeholder', () => {
  it('should render', () => {
    const tree = renderer.create(
      <Providers>
        <Typography typographicClassName={'h1'}>
          <H1>Boom</H1>
        </Typography>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render children', () => {
    const tree = renderer.create(
      <Providers>
        <Typography isLoading={false} typographicClassName={'h1'}>
          <H1>Boom</H1>
        </Typography>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
