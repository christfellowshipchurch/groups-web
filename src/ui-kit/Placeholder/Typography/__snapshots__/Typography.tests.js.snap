// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Typography placeholder should render 1`] = `
<mark
  className="ra2rjz-0 gJMToV h1 ui-kit.Placeholder.Line custom-placeholder"
  width="100%"
/>
`;

exports[`The Typography placeholder should render children 1`] = `
<h1
  className="sc-1dj0tjq-0 eKKMjd ui-kit-typography-H1-StyledH1"
>
  Boom
</h1>
`;
